### ISSUE TITLE
starting from systemBinary does not work as documented

### ISSUE THREAD

**Issue by boredland** [nodkz/mongodb-memory-server#511]
## Versions

- NodeJS: v14.16.1
- mongodb-memory-server-*: 7.2.1
- mongodb(the binary version): 4.4.5
- system: Manjaro Linux

package: mongo-memory-server

## What is the Problem?

Starting the mongo memory server with

```js
{
  instance: { port: 27017, dbPath: undefined },
  binary: { version: '4.4.7', systemBinary: '/usr/bin/mongod' }
}
```
or 
```js
{
  instance: { port: 27017, dbPath: undefined },
  binary: { systemBinary: '/usr/bin/mongod' }
}
```

leads the memory server to download and use the downloaded binary (4.4.7, 4.0.25 respectively), while it should use the specified one.

if im setting the environment variable `MONGOMS_SYSTEM_BINARY=/usr/bin/mongod`, the system binary is being used, but the output is:

```
getPath: MongoMemoryServer: Possible version conflict
  SystemBinary version: "4.4.5"
  Requested version:    "4.0.25"

  Using SystemBinary!
```

which in my eyes means that the start works as expected, but the output is kinda awkward because explicitly did not want to start 4.0.25, but the system binary. The output is a minor issue I don't really care about - the 'systemBinary' having no effect is problematic.

**Comment by hasezoey** [comment]
please provide [debug output](https://github.com/nodkz/mongodb-memory-server#enable-debug-mode)
and also how you try to start an instance (like `new MongoMemoryServer({ optionHere })`)

> which in my eyes means that the start works as expected, but the output is kinda awkward because explicitly did not want to start 4.0.25, but the system binary.

this is just to ensure the correct version is started, the version can be set in like the `package.json`, but the systemBinary can be an env variable individual to each system, and to notify an version conflict (and the default is `4.0.25` currently)

PS: if you want to disable runtime download, you can set the option [`RUNTIME_DOWNLOAD`](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options#runtime_download) to `false` (also read [package.json config](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options#how-to-use-them-in-the-packagejson) and [environment config](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options#how-to-use-them-as-environment-variables))

**Comment by boredland** [comment]
I reduced my implementation for the sake of this example a bit, but ran it now like this:

```
const mongod = new MongoMemoryServer({
      instance: { port: 27017 },
      binary: { systemBinary: '/usr/bin/mongod' },
});
await mongod.start();
```

```
$ node -r dotenv-flow/config -r ts-node/register/transpile-only -r tsconfig-paths/register src/utils/start-mongo.ts
  MongoMS:ResolveConfig findPackageJson: Found package.json at "/home/jonass/Documents/***/packages/gql-server/package.json" +0ms
  MongoMS:ResolveConfig findPackageJson: Found package.json at "/home/jonass/Documents/***/package.json" +3ms
  MongoMS:MongoMemoryServer start: Called .start() method +0ms
  MongoMS:MongoMemoryServer _startUpInstance: Called MongoMemoryServer._startUpInstance() method +1ms
  MongoMS:MongoMemoryServer getStartOptions +1ms
  MongoMS:MongoMemoryServer _startUpInstance: Creating new MongoDB instance with options: {"instance":{"port":27017,"dbName":"","ip":"127.0.0.1","storageEngine":"ephemeralForTest","dbPath":"/tmp/mongo-mem--14952-RClp3y0ABfa7","tmpDir":{"name":"/tmp/mongo-mem--14952-RClp3y0ABfa7"}},"binary":{"systemBinary":"/usr/bin/mongod"}} +4ms
  MongoMS:MongoInstance create: Called .create() method +0ms
  MongoMS:MongoInstance Mongo[27017]: start +1ms
  MongoMS:MongoBinary getPath +0ms
  MongoMS:utils tryReleaseFile: "/etc/upstream-release/lsb-release" does not exist +0ms
  MongoMS:getos getLinuxInformation: Using etcOsRelease +0ms
  MongoMS:DryMongoBinary generateDownloadPath: Generating Download Path, preferGlobal: "true" +0ms
  MongoMS:DryMongoBinary generateDownloadPath: Paths: {
  legacyHomeCache: '/home/jonass/.cache/mongodb-binaries/mongod-x64-manjaro-4.0.25',
  modulesCache: '/home/jonass/Documents/sputnik/packages/gql-server/node_modules/.cache/mongodb-memory-server/mongod-x64-manjaro-4.0.25',
  relative: '/home/jonass/Documents/sputnik/packages/gql-server/mongodb-binaries/mongod-x64-manjaro-4.0.25',
  resolveConfig: ''
} +2ms
  MongoMS:DryMongoBinary generateDownloadPath: Found binary in legacyHomeCache: "/home/jonass/.cache/mongodb-binaries/mongod-x64-manjaro-4.0.25" +6ms
  MongoMS:MongoBinary getPath: MongoBinary options: {
  "version": "4.0.25",
  "downloadDir": "/home/jonass/.cache/mongodb-binaries",
  "os": {
    "os": "linux",
    "dist": "manjaro",
    "release": "",
    "id_like": "arch"
  },
  "arch": "x64",
  "systemBinary": "",
  "platform": "linux",
  "checkMD5": false
} +17ms
  MongoMS:DryMongoBinary locateBinary: Trying to locate Binary for version "4.0.25" +1ms
  MongoMS:DryMongoBinary generateDownloadPath: Generating Download Path, preferGlobal: "true" +1ms
  MongoMS:DryMongoBinary generatePaths: resolveConfigValue is not empty +0ms
  MongoMS:DryMongoBinary generateDownloadPath: Paths: {
  legacyHomeCache: '/home/jonass/.cache/mongodb-binaries/mongod-x64-manjaro-4.0.25',
  modulesCache: '/home/jonass/Documents/sputnik/packages/gql-server/node_modules/.cache/mongodb-memory-server/mongod-x64-manjaro-4.0.25',
  relative: '/home/jonass/Documents/sputnik/packages/gql-server/mongodb-binaries/mongod-x64-manjaro-4.0.25',
  resolveConfig: '/home/jonass/.cache/mongodb-binaries/mongod-x64-manjaro-4.0.25'
} +1ms
  MongoMS:DryMongoBinary generateDownloadPath: Found binary in resolveConfig (DOWNLOAD_DIR): "/home/jonass/.cache/mongodb-binaries/mongod-x64-manjaro-4.0.25" +0ms
  MongoMS:DryMongoBinary locateBinary: running generateDownloadPath +0ms
  MongoMS:DryMongoBinary generateDownloadPath: Generating Download Path, preferGlobal: "true" +1ms
  MongoMS:DryMongoBinary generatePaths: resolveConfigValue is not empty +0ms
  MongoMS:DryMongoBinary generateDownloadPath: Paths: {
  legacyHomeCache: '/home/jonass/.cache/mongodb-binaries/mongod-x64-manjaro-4.0.25',
  modulesCache: '/home/jonass/Documents/sputnik/packages/gql-server/node_modules/.cache/mongodb-memory-server/mongod-x64-manjaro-4.0.25',
  relative: '/home/jonass/Documents/sputnik/packages/gql-server/mongodb-binaries/mongod-x64-manjaro-4.0.25',
  resolveConfig: '/home/jonass/.cache/mongodb-binaries/mongod-x64-manjaro-4.0.25'
} +0ms
  MongoMS:DryMongoBinary generateDownloadPath: Found binary in resolveConfig (DOWNLOAD_DIR): "/home/jonass/.cache/mongodb-binaries/mongod-x64-manjaro-4.0.25" +1ms
  MongoMS:DryMongoBinary locateBinary: found binary at "/home/jonass/.cache/mongodb-binaries/mongod-x64-manjaro-4.0.25" +0ms
  MongoMS:MongoBinary getPath: Mongod binary path: "/home/jonass/.cache/mongodb-binaries/mongod-x64-manjaro-4.0.25" +5ms
```

**Comment by boredland** [comment]
> PS: if you want to disable runtime download, you can set the option RUNTIME_DOWNLOAD to false (also read package.json config and environment config)

this isn't sth. I want to achieve. My goal (and setup working until the major update) was to use the local binary if it exists and download if it doesn't. This worked previously using [lookpath](https://www.npmjs.com/package/lookpath) to find the mongod binary and when systemBinary was undefined, it downloaded one.

**Comment by hasezoey** [comment]
this issue may be fixed with 7.3.1, which should be published in some minutes

**Comment by boredland** [comment]
tyvm, works like a charm now!
